yaksh/test_views.py

```diff
@@ -6286,3 +6288,226 @@ def test_preview_lesson_description(self):
         self.assertEqual(response.json()['data'], '<p>test description</p>')
+
+
+class TestPost(TestCase):
+    def setUp(self):
+        self.client = Client()
+        self.mod_group = Group.objects.create(name='moderator')
+
+        self.student_plaintext_pass = 'student'
+        self.student = User.objects.create_user(
+            username='student',
+            password=self.student_plaintext_pass,
+            first_name='first_name',
+            last_name='last_name',
+            email='student@test.com'
+        )
+
+        Profile.objects.create(
+            user=self.student,
+            roll_number=10,
+            institute='IIT',
+            department='Chemical',
+            position='student',
+            timezone='UTC'
+        )
+
+        # moderator
+        self.user_plaintext_pass = 'demo'
+        self.user = User.objects.create_user(
+            username='demo_user',
+            password=self.user_plaintext_pass,
+            first_name='first_name',
+            last_name='last_name',
+            email='demo@test.com'
+        )
+
+        Profile.objects.create(
+            user=self.user,
+            roll_number=10,
+            institute='IIT',
+            department='Chemical',
+            position='Moderator',
+            timezone='UTC'
+        )
+
+        self.course = Course.objects.create(
+            name="Python Course",
+            enrollment="Enroll Request", creator=self.user
+        )
+
+    def test_view_course_forum_denies_anonymous_user(self):
+        url = '/exam/login/?next=/exam/forum/' + str(self.course.id) + '/'
+        response = self.client.get(reverse('yaksh:course_forum', kwargs={
+            'course_id': self.course.id
+        }), follow=True)
+        self.assertEqual(response.status_code, 200)
+        redirection_url = url
+        self.assertRedirects(response, redirection_url)
+
+    def test_view_course_forum(self):
+        self.client.login(
+            username=self.student.username,
+            password=self.student_plaintext_pass
+        )
+        response = self.client.get(reverse('yaksh:course_forum', kwargs={
+            'course_id': self.course.id
+        }), follow=True)
+        self.assertEquals(response.status_code, 200)
+        self.assertTemplateUsed(response, 'yaksh/course_forum.html')
+
+    def test_create_post(self):
+        self.client.login(
+            username=self.student.username,
+            password=self.student_plaintext_pass
+        )
+        response = self.client.post(reverse('yaksh:course_forum', kwargs={
+            'course_id': self.course.id
+        }), {
+            'title': 'post 1',
+            'description': 'post 1 description',
+            'course': self.course,
+            'creator': self.student
+        })
+        self.assertEquals(response.status_code, 302)
+        url = response.url.split('/')
+        uid = url[5]
+        test_against = Post.objects.get(uid=uid)
```

Comment:
Instead of getting uid from URL and getting post object, you can directly check `self.assertTrue(Post.objects.filter(title='post 1', creator=self.student, course=self.course).exists())`